extensions/identity-hub/src/main/java/org/eclipse/dataspaceconnector/identityhub/IdentityHubExtension.java

```diff
@@ -49,12 +63,38 @@ public void initialize(ServiceExtensionContext context) {
         methodProcessorFactory.register(COLLECTIONS_WRITE, new CollectionsWriteProcessor(identityHubStore));
         methodProcessorFactory.register(FEATURE_DETECTION_READ, new FeatureDetectionReadProcessor());
 
-        var identityHubController = new IdentityHubController(methodProcessorFactory);
+        var path = context.getSetting(SELF_DESCRIPTION_DOCUMENT_PATH_SETTING, null);
+        JsonNode selfDescription;
+        try (var is = loadSelfDescriptionDocument(path)) {
```

Comment:
I would move the whole logic to load the self description as a JsonNode into the method, or even better, into a collaborator `SelfDescriptionLoader` that can be unit tested?